index.js

```diff
@@ -99,10 +129,10 @@ function compatibleMessage(thrown, errMatcher) {
 function getConstructorName(errorLike) {
   var constructorName = errorLike;
   if (errorLike instanceof Error) {
-    constructorName = errorLike.constructor.name;
+    constructorName = getFunctionName(errorLike.constructor);
   } else if (typeof errorLike === 'function') {
     // if `err` is not an instance of Error it is an error constructor itself
-    constructorName = new errorLike().name; // eslint-disable-line new-cap
+    constructorName = getFunctionName(errorLike); // eslint-disable-line new-cap
```

Comment:
You seem to be right.
Apparently we didn't need `new` at all. I've just tested this with and without `new` and results are the same.